### PR TITLE
Cleanup ROOT recipe

### PR DESCRIPTION
**Pull request by ktf**
* Drop support for ROOT5
* Add support for C++20
* Make sure the plugins / paths are set up correctly in the system.rootrc also in the case the incremental recipe is used.

### CODE REVIEW

File: root.sh [hunk 1]
```diff
@@ -130,6 +135,7 @@ cmake $SOURCEDIR
       ${CXX11:+-Dcxx11=ON}                                                             \
       ${CXX14:+-Dcxx14=ON}                                                             \
       ${CXX17:+-Dcxx17=ON}                                                             \
+      ${CXX17:+-Dcxx20=ON}                                                             \
```
Comment (TimoWilken) [inline root.sh:138]: Why keep the `CXXvv` variables? You removed the code that sets them.

If you keep them, shouldn't the last one be `${CXX20:+-Dcxx20=ON}` instead?

File: root.sh [hunk 2]
```diff
@@ -34,26 +34,31 @@ incremental_recipe: |
...
          "$INSTALLROOT/etc/plugins/TFile/P070_TAlienFile.C"
+  # Add support for ROOT_PLUGIN_PATH envvar for specifying additional plugin search paths
+  grep -v '^Unix.*.Root.PluginPath' $INSTALLROOT/etc/system.rootrc > system.rootrc.0
+  cat >> system.rootrc.0 <<\EOF
+  # Specify additional plugin search paths via the environment variable ROOT_PLUGIN_PATH.
+  # Plugins in \$ROOT_PLUGIN_PATH have priority.
+  Unix.*.Root.PluginPath: $(ROOT_PLUGIN_PATH):$(ROOTSYS)/etc/plugins:
+  Unix.*.Root.DynamicPath: .:$(ROOT_DYN_PATH):
+  EOF
+  mv system.rootrc.0 $INSTALLROOT/etc/system.rootrc
```
Comment (TimoWilken) [inline root.sh:45]: Why not quote `$INSTALLROOT`?

Comment (ktf) [reply root.sh:45]:
```suggestion
  # Add support for ROOT_PLUGIN_PATH envvar for specifying additional plugin search paths
  grep -v '^Unix.*.Root.PluginPath' "$INSTALLROOT/etc/system.rootrc" > system.rootrc.0
  cat >> system.rootrc.0 <<\EOF
  # Specify additional plugin search paths via the environment variable ROOT_PLUGIN_PATH.
  # Plugins in \$ROOT_PLUGIN_PATH have priority.
  Unix.*.Root.PluginPath: $(ROOT_PLUGIN_PATH):$(ROOTSYS)/etc/plugins:
  Unix.*.Root.DynamicPath: .:$(ROOT_DYN_PATH):
  EOF
  mv system.rootrc.0 "$INSTALLROOT/etc/system.rootrc"
```

File: root.sh (hunk 1 above)
Comment (ktf) [reply root.sh:138]:
```suggestion
```

**Comment by TimoWilken** [comment]
@ktf caching is done (for both PRs together). Do you want to merge them today or on Monday?

**Pull request merged**